### PR TITLE
Updated to fix macOS errors with changes to Process.launch() in 10.13

### PR DESCRIPTION
**Pull request by bdrelling**
**Description**
- Raised minimum `macOS` platform version to `.v10_13`. High Sierra released in 2017 and can run on devices as far back as 2014 or so, so this feels very reasonable.
- Added `tmp` to `.gitignore`, on the off chance the folder hangs around locally when testing, such as in the event of a `zinc` error.
- Updated Mintfile.
- Abstracted the creation and running of `Process` into `ConfiguredProcess`. Splitting this into to functions (`init` and `run`) helped make the logic a lot easier.
- Renamed `ShellRunner` to `ProcessRunner` to better match what the file did, which was more than just shell commands. I'm not sure why the diff dropped, but it was the same rename as anything else.
- Added the `Executable` enum, which extends upon the previous `Shell` enum.
- Fixed deprecations and re-indented a function in `SyncCommand`.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -6,7 +6,7 @@ import PackageDescription
 let package = Package(
     name: "Zinc",
     platforms: [
-        .macOS(.v10_10),    // minimum supported version via SPM
+        .macOS(.v10_13),    // minimum supported version via SPM
```
Comment (khaptonstall) [inline Package.swift:9]: If the comment is no longer true, can that be removed?

File: Sources/Libraries/ZincFramework/Sync/SyncSubcommand.swift [hunk 2]
```diff
@@ -14,12 +14,12 @@ struct SyncSubcommand: ParsableCommand {
     // MARK: Options
     
     /// The Zincfile to parse.
-    @Option(name: .shortAndLong, default: "Zincfile", help: "The Zincfile to parse and use for syncing.")
-    private var file: String
+    @Option(name: .shortAndLong, help: "The Zincfile to parse and use for syncing.")
+    private var file: String = "Zincfile"
```
Comment (khaptonstall) [inline Sources/Libraries/ZincFramework/Sync/SyncSubcommand.swift:18]: Q: By removing the `default` argument, does the help page for this command now lose that information? If so, I feel like the preferred route would be to keep it as an argument in the property wrapper.

File: Sources/Libraries/CarbonFramework/ConfiguredProcess.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        if #available(macOS 10.15.4, *) {
+            outputData = try self.outputPipe.fileHandleForReading.readToEnd() ?? Data()
+        } else {
+            outputData = self.outputPipe.fileHandleForReading.readDataToEndOfFile()
+        }
+        
+        // Convert data to strings.
+        let output = String(data: outputData, encoding: .utf8)
+        
+        // If there's no putput, throw an error
```
Comment (khaptonstall) [inline Sources/Libraries/CarbonFramework/ConfiguredProcess.swift:50]: 🏌️ ⛳ 
```suggestion
        // If there's no output, throw an error
```

**Review by khaptonstall — CHANGES_REQUESTED** [review]

File: Sources/Libraries/ZincFramework/Sync/SyncSubcommand.swift (hunk 2 above)
Comment (bdrelling) [reply Sources/Libraries/ZincFramework/Sync/SyncSubcommand.swift:18]: Help page should be identical, these changes are the result of deprecations. The Property Wrapper can have access to the underlying default value, so I assume that's why they removed it and required this.

**Comment by bdrelling** [comment]
@khaptonstall should be ready for re-review!

As a note, I have no idea why CI is failing specifically, but at this point any continued work I'm going to do in `Zinc` will be at `bdrelling/Zinc` (because not having administration over a repo I want to support is kind of a no-go), and CI is working on that branch.

Let me know if SpotHero wants to keep maintaining the repo, or if y'all just want to point to my repo from the `Mintfile`. As a build tool, I imagine this is less concerning since you could rip it out at any point.

**Comment by bdrelling** [comment]
Also, as a follow-up note: I think you'll have to be the person who merges it since I'm just a lowly contributor.

Feel free to take this PR over for anything else that needs changing, I just didn't want to leave `spothero/Zinc` hanging without a fix, as I promised. :)

**Review by khaptonstall — DISMISSED** [review]

**Comment by khaptonstall** [comment]
@bdrelling 
> Feel free to take this PR over for anything else that needs changing, I just didn't want to leave spothero/Zinc hanging without a fix, as promised. 

Thanks! I'll try to dig into failing CI first to see what needs to be done to get it green and then I'll merge.

**Comment by bdrelling** [comment]
@khaptonstall sounds good! I invited you to `bdrelling/zinc` and plan to put some more improvements on that repo as well, including building out some actual tests. I will likely not add `danger` back in for a little while, as a heads up.

One thing I'll commit back as well: the `--enable-test-discovery` flag is no longer required, but also completely broken and is unable to catch failing tests if left alone. It took me _hours_ to figure out that this was the culprit to some other errors.

**Comment by bdrelling** [comment]
I'll let you know when `bdrelling/zinc` is in better working order and you can make the call on switching the source for any `Mintfile` and so on. Happy to make anyone from SpotHero iOS a maintainer on it, but I think most of y'all didn't ever touch it, so I just added you for now.

**Comment by khaptonstall** [comment]
@bdrelling Sounds good. Yeah, let me think on how we want to move forward with our integration of Zinc internally and I'll get back to you!

**Pull request merged**